packages/phoenix-event-display/src/managers/three-manager/import-manager.ts

```diff
@@ -208,29 +208,47 @@ export class ImportManager {
   }
 
   /**
-   * Loads a GLTF (.gltf) scene/geometry from the given URL.
+   * Loads a GLTF (.gltf) scene(s)/geometry from the given URL.
    * @param sceneUrl URL to the GLTF (.gltf) file.
-   * @param name Name of the loaded scene/geometry.
-   * @param callback Callback called after the scene/geometry is loaded.
+   * @param name Name of the loaded scene/geometry if a single scene is present, ignored if several scenes are present
+   * @param menuNodeName Name of the menu where to add the scene in the gui
+   * @param callback Callback called after the each scene/geometry is loaded.
    * @param scale Scale of the geometry.
    * @returns Promise for loading the geometry.
    */
   public loadGLTFGeometry(
     sceneUrl: string,
     name: string,
-    callback: (Geometry: Object3D) => any,
-    scale?: number
+    menuNodeName: string,
+    callback: (Geometry: Object3D,
+               geoName: string,
+               menuName: string,
+               visible: boolean) => any,
+    scale: number,
+    initiallyVisible: boolean
   ): Promise<unknown> {
     const loader = new GLTFLoader();
     return new Promise<void>((resolve, reject) => {
       loader.load(
         sceneUrl,
         (gltf) => {
-          const geometry = gltf.scene;
-          this.processGeometry(geometry, name, scale);
-          callback(geometry);
-          resolve();
-          this.loadingManager.itemLoaded(`gltf_geom_${name}`);
+          for (const geometry of gltf.scenes) {
+            var geoName = (name === "")? geometry.name : name;
+            // handle names with ' > ', these include paths which should go to menuName
+            const pos = geoName.lastIndexOf('_>_');
+            var menuName = menuNodeName;
+            if (pos > -1) {
+              if (menuName != "") menuName = menuName + ' > ';
+              menuName = menuName + geoName.substring(0, pos).replace('_>_', ' > ');
+              geoName = geoName.substring(pos + 3);
+            }
+            this.processGeometry(geometry, geoName, scale);
+            var visible = initiallyVisible;
+            if ("visible" in geometry.userData) visible = geometry.userData["visible"];
+            callback(geometry, geoName, menuName, visible);
```

Comment:
As I understand it, all the scenes will be loaded as separate geometries? The geometries are then added to the menu separately as well?

Reply:
Could you also please tell me in what format the geometry names are set in the glTF file? The separator seems to be `_>_`.

Reply:
I think there may be a better way to handle this instead of going through all these callbacks which makes the code hard to read and understand.

I can think of something once I get an answer to the questions above.

Reply:
Yes, all scenes will be loaded as separate geometries, very much like it's done now actually, except we allow to have them all in a single file and do not force users to have n files, one per geometry. As gltf format is having this functionality, it would be sad not to use it.

Concerning format, I've just kept what was used so far, so ' > ' separator. The thing however is that spaces in the gltf file seem to be replaced by underscores at some level, so it ends up being "_>_". No clue why.

Concerning callbacks, I support your position, but again tried to follow the style of what was there, that is using callbacks everywhere. Seems to be also a generic thing in javascript.